components/layermanager/layer-selector.service.ts

```diff
@@ -0,0 +1,16 @@
+
+import {Injectable} from '@angular/core';
+import {Layer} from 'ol/layer';
+import {Subject} from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class HsLayerSelectorService {
+  layerSelected: Subject<any> = new Subject();
+  constructor() {}
+
+  select(layer: Layer) {
```

Comment:
Is the type right? Shouldn't it be HsLayerDescriptor instead of ol/Layer, since e are calling layer.layer later on it. Also Subject<HsLayerDescriptor>

Reply:
This can get very confusing quite quickly. I suspect better variable name (layerDesc or alike) would make the code much clearer 😉